tools/segment.sh

```diff
@@ -0,0 +1,109 @@
+filename="Forbidden-Mixed-MP4"
+relaxations="Rfe Fre Wse Hat $file"
+
+# make file into array and initialise variables
+IFS=' ' read -r -a array <<< "$file"
+stringRelax=()
+temp=""
+
+# create new array that respects square brackets
+for one_thing in "${array[@]}"; do
+    if [ "${one_thing: -1}" = "," ]
+    then
+        temp+="$one_thing"
+    else
+        if [ -z "$temp" ]
+        then
+            stringRelax+=("$one_thing")
+        else
+            stringRelax+=("$temp$one_thing")
+            temp=""
+        fi
+    fi
+done
+
+# randomise function
+shuffle() {
+   local i tmp size max rand
+
+   # $RANDOM % (i+1) is biased because of the limited range of $RANDOM
+   # Compensate by using a range which is a multiple of the array size.
+   size=${#array2[*]}
+   max=$(( 32768 / size * size ))
+
+   for ((i=size-1; i>0; i--)); do
+      while (( (rand=$RANDOM) >= max )); do :; done
+      rand=$(( rand % (i+1) ))
+      tmp=${array2[i]} array2[i]=${array2[rand]} array2[rand]=$tmp
+   done
+}
+
+# do the shuffling
+array2=("${stringRelax[@]}")
+shuffle
+
+rm -rf src && mkdir src
+total_size=${#array2[@]}
+((relax_size=$total_size/$seg_size))
+
+
+if [ $oneloc -eq 1 ]
+then
+    ol="-oneloc"
+else
+    ol=""
+fi
+
+
+
+
+# create the config files
+whole=""
+for ((i=0; i<${seg_size}; i++)); do
+    cd src && rm -rf src${i} && mkdir src${i} && cd ..
+    for aspect in "${array2[@]:(i*${relax_size}+1):${relax_size}}"; do
+        whole+="$aspect "
+    done
+    echo "-arch AArch64
+
+-nprocs $nprocs
+-eprocs
+-size $csize
+
+$ol
+
+-name Armv8-MIXED-ext-forbidden-$i-
+
+-relaxlist Rfe Fre Wse Hat $whole" >./src/src${i}/${filename}${i}.conf
+    whole=""
+done
+
+# run diy7
+for ((i=0; i<${seg_size}; i++)); do
+    if [ $timeout -eq 0 ]
+    then
+        diy7 -o ./src/src${i} -conf ./src/src${i}/${filename}${i}.conf
+    else
+        diy7 -o ./src/src${i} -conf ./src/src${i}/${filename}${i}.conf& read -t $timeout || kill $!
+    fi
+done
+
+
+
+for ((i=0; i<${seg_size}; i++)); do
+    echo "./src/src${i}/@all" >>@all
+    echo "\n" >>@all
```

Comment:
This lines introduces untimely `\` `n` sequences in the output file `@all`. Deleting it apparently solves the problem: 
```suggestion
```